src/core/pointcloud/qgspointcloudrenderer.cpp

```diff
@@ -159,6 +169,37 @@ bool QgsPointCloudLayerRenderer::render()
   return true;
 }
 
+QList<IndexedPointCloudNode> QgsPointCloudLayerRenderer::traverseTree( const QgsPointCloudIndex *pc,
+    const QgsRenderContext &context,
+    IndexedPointCloudNode n,
+    float maxErrorPixels,
+    float nodeErrorPixels )
+{
+  QList<IndexedPointCloudNode> nodes;
+
+  if ( context.renderingStopped() )
+  {
+    qDebug() << "canceled";
+    return nodes;
+  }
```

Comment:
no need to pass context to this method and check context.renderingStopped() here - the traversal is very fast